### ISSUE TITLE
Use QUIC as transport protocol

### ISSUE THREAD

**Issue by anvilvapre** [OpenRA/OpenRA#20223]
QUIC is a UDP based protocol that offer rialable transportation of data. HTTP3 uses it.

An advantage is that it uses connection ID's over IP adresses plus port number to indentify a connection. I.e. when a user switches from a mobile connection to Wifi, or switches Wifi access points - the stream can continue where it left off - following a different path. 

Feature request:
- Consider running game protocol over QUIC.
- Or, consider adding a connection ID to the protocol. Allowing user within i.e. the time frame of 60 seconds to reconnect to a running game - using a new tcp/ip connection. 

Benefit:
- Multi player games can continue after the TCP connection of one client drops. 
- People have an oppertunity to switch from Wifi to Cable - in gane.
- With a force reconnexct button under settings. Players can do a quick reconnect - where a new TCP connection may have less lag.

**Comment by teinarss** [comment]
I had the idea to replace the TCP implementation with a reliable UDP
library, see https://github.com/OpenRA/OpenRA/pull/20119

Last time I read some articles around using quic for games it looked that
it was not as optimal as using a simpler reliable UDP. And there May be
some problems with the cross platform implementation on .net.

See
https://robertsspaceindustries.com/spectrum/community/SC/forum/50259/thread/network-improvements/2689330

**Comment by anvilvapre** [comment]
Yeah, for all languages, the number of implementations of QUIC is still limited. 

Not sure what benefit UDP over TCP would have. Also because the games syncs each frame. In the end the client will need to always receive that last packet. I would assume you will end up reimplementing TCP.

In the past Unreal (1) and I believe also Quake would allow certain fields to be marked with a prio. If an packet containing low prio data would get lost it would be just be dropped and not resent. So assume UDP had more advantage here. Not sure if UDP gives you more control over when packets are sent/flushed by the OS. 

Using UDP would likely get rid of the issue of a lost/resetted TCP connections. Allowing 'reconnects'.

**Comment by teinarss** [comment]
See https://gafferongames.com/post/deterministic_lockstep/

**Comment by anvilvapre** [comment]
https://learn.microsoft.com/en-us/dotnet/fundamentals/networking/quic/quic-overview
